### PR TITLE
Adding Support of Raft Integrated Storage as Default Option

### PR DESCRIPTION
**Pull request by nehrman**

### CODE REVIEW

File: templates/vault_backend_raft.j2 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+backend "raft" {
+  path = "{{ vault_raft_data_path }}"
+  node_id = "{{ vault_raft_node_id }}"
+  {% for server in groups['vault_raft_servers'] | difference([inventory_hostname]) %}
+    {% if not (vault_tls_disable | bool) %}
+  retry_join {
+    leader_api_addr =  "https://{{ hostvars[server]['ansible_'+vault_iface]['ipv4']['address'] }}:{{ vault_port }}"
+    leader_ca_cert = "{{ vault_tls_config_path }}/{{ hostvars[server]['vault_tls_ca_file'] }}"
```
Comment (kvonbredow) [inline templates/vault_backend_raft.j2]: This should probably be replaced with `{{ vault_backend_tls_config_path }}/{{ vault_backend_tls_ca_file }}` to match the etcd implementation

Same goes for the next two lines

**Comment by bbaassssiiee** [comment]
Please resolve the conflicts

File: defaults/main.yml [hunk 2]
```diff
@@ -62,14 +62,14 @@ vault_main_configuration_template: vault_main_configuration.hcl.j2
...
 
-vault_backend: consul
 vault_backend_consul: vault_backend_consul.j2
 vault_backend_file: vault_backend_file.j2
 vault_backend_raft: vault_backend_raft.j2
 vault_backend_etcd: vault_backend_etcd.j2
 vault_backend_s3: vault_backend_s3.j2
 vault_backend_dynamodb: vault_backend_dynamodb.j2
 vault_backend_mysql: vault_backend_mysql.j2
+vault_backend_raft: vault_backend_raft.j2
```
Comment (eagleusb) [inline defaults/main.yml]: **issue**: you seem to have a duplicate of line 67

**Comment by eagleusb** [comment]
@nehrman Are you still on it ? That would be an interesting change to the role. :+1:

File: defaults/main.yml (hunk 2 above)
Comment (nehrman) [reply defaults/main.yml]: SHoud be good now.

**Review by nehrman — COMMENTED** [review]
Done

**Comment by willhaines** [comment]
@ksatirli Does this need anything else before merging?

**Comment by ksatirli** [comment]
This looks good to me, I'm especially liking the `disable_mlock` option. 

Thank you for your work on this!

Tagging in @bbaassssiiee for final once-over.

**Review by ksatirli — APPROVED** [review]

**Comment by bbaassssiiee** [comment]
Well, this is a big change, Vault can stand on its own feet. Be careful with updates where you have Consul or other backends.

**Review by bbaassssiiee — APPROVED** [review]

**Pull request merged**